test/e2e/parallel/space_test.go

```diff
@@ -429,30 +429,34 @@ func TestSpaceNamespaces(t *testing.T) {
 	t.Parallel()
 	// make sure everything is ready before running the actual tests
 	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
 	memberAwait := awaitilities.Member1()
 
 	// when
 	space, _, _ := CreateSpace(t, awaitilities, WithTierName("base"), WithTargetCluster(memberAwait.ClusterName))
 	// then
 	// wait until Space has been provisioned on member-1
 	space, nsTmplSet := VerifyResourcesProvisionedForSpace(t, awaitilities, space.Name, UntilSpaceHasStatusTargetCluster(memberAwait.ClusterName))
+	expectedProvisionedNamespaces := []toolchainv1alpha1.SpaceNamespace{
+		{
+			Name: space.Name + "-dev",
+			Type: "default", // default ns name should be there
+		},
+		{
+			Name: space.Name + "-stage",
+		},
+	}
 
 	// then
 	// wait until NSTemplateSet has expected list of provisioned namespaces
-	// todo check also list of provisioned namespaces from space once functionality will be available
 	_, err := memberAwait.WaitForNSTmplSet(t, nsTmplSet.Name,
```

Comment:
Do we really care about verifying the NSTemplateSets here?
Isn't the NSTemplateSet is just an implementation detail?
What we need is to make sure:
- All the end resources are provisioned for the space (Namespaces with all expected resources there, cluster scoped resources, etc). And this is what is already done in line 439 (`VerifyResourcesProvisionedForSpace()`), right?
- Check the Space status including the list of namespaces.

It doesn't heart to check the NsTemplateSet too but do we need it in our e2e tests for Spaces?
WDYT?

Reply:
Would it make sense to add a check in https://github.com/codeready-toolchain/toolchain-e2e/blob/master/testsupport/tiers/nstemplateset.go#L17 instead so that anything that calls the `VerifyNSTemplateSet` func will check the provisioned namespaces status too? That function has the slice of namespaces `expectedTemplateRefs.Namespaces` already so it should be easy to verify there?

Reply:
You are right, I simplified (hopefully) a lot the version of the tests, please see: [3a8e178](https://github.com/codeready-toolchain/toolchain-e2e/pull/667/commits/3a8e17812b60eafa5d2891bafce9f933a3dc9e55)

Basically as @rajivnathan suggested (thanks for this, haven't noticed that function that returns the list of namespaces provisioned from the tier template), now I check the provisioned namespaces in `expectedTemplateRefs.Namespaces` in the `VerifyNSTemplateSet` function , and I've removed all the "hard coded" checks for provisioned namesapces in all other places.

So in brief now,  if I did it correctly, `VerifyResourcesProvisionedForSpace` function checks the provisioned namespaces for both the `NSTemplateSet` and the `Space` , it checks this list against the namespaces that are coming from the tier template. No need to check those in the individual tests in theory. For the same reason I've deleted the previously created test `TestSpaceNamespaces`.

Thanks for your suggestions and let me know WDYT!